parcels/particleset.py

```diff
@@ -520,7 +521,7 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
             walltime_start = time_module.time()
         if verbose_progress:
             pbar = self.__create_progressbar(_starttime, endtime)
-        while (time < endtime and dt > 0) or (time > endtime and dt < 0) or dt == 0:
+        while ((time < endtime and dt > 0) or (time > endtime and dt < 0)) and not self.aborted or dt == 0:
```

Comment:
So this `self.aborted` is only used in the unit test? Do we then really need it?